src/Libraries/Watch3DNodeModelsWpf/HelixWatch3DNodeViewModel.cs

```diff
@@ -74,9 +76,49 @@ protected override void OnModelPropertyChanged(object sender, PropertyChangedEve
             }
         }
 
+        protected override void OnNodePropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            var node = sender as NodeModel;
```

Comment:
I think you can make this a bit simpler by using:
(if sender is NodeModel node){
}